android/source/src/main/kotlin/com/gu/source/components/chips/ChipDecoration.kt

```diff
@@ -0,0 +1,212 @@
+package com.gu.source.components.chips
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.gu.source.R
+import com.gu.source.Source
+import com.gu.source.components.chips.ChipDecoration.Icon
+import com.gu.source.components.chips.ChipDecoration.Image
+import com.gu.source.icons.Check
+
+/**
+ * Represents an icon or image displayed before or after a [SourceChip]'s text.
+ *
+ * [Icon]s are `18.dp` tall and the [Image]s are `24.dp` tall.
+ *
+ * The icon or image will be tinted with the [SourceChip.Style.contentColour]. To override the
+ * tinting, use [ChipDecoration.Icon.Component] or [ChipDecoration.Image.Component] and provide a
+ * custom composable function with the desired colour.
+ *
+ * For icons, prefer to use [ChipDecoration.Icon.Vector] with a [Source.Icons] vector where possible.
+ * Alternatively use [ChipDecoration.Icon.Painter] with a drawable resource ID.
+ *
+ * For images, prefer to use [ChipDecoration.Image.Painter] for static images. For images fetched
+ * from the network, use [ChipDecoration.Image.Component] with a custom composable function.
+ */
+sealed class ChipDecoration {
```

Comment:
Great improvement, this make the chip much more flexibility.maybe we can add  alignment options  and shape later.